### ISSUE TITLE
Domain request table accessibility issues - ANDI + VO output

### ISSUE THREAD

**Issue by asaki222** [cisagov/manage.get.gov#3048]
### Issue description
This ticket outlines accessibility issues with the Domain Request table, both on portfolios and for non-portfolios found during #2594.

### Acceptance criteria

- [ ] All sub issues are completed
- [ ] On the domain request table, any glaring accessibility issues found but not otherwise listed are also resolved. 
    - Only attempt to fix items that would otherwise be listed as a "quick fix" if actualized into a ticket. The time investment on this particular AC should not exceed one hour.

### Additional context

All tickets in this issue have originated from this [slack channel](https://dhscisa.enterprise.slack.com/archives/C07TZN9MQ9Z).
Pointing as a three due to the testing requirement and potential complexities due to that.

### Links to other issues
Relates to #2594
